### PR TITLE
Watch to run ghcid

### PR DESCRIPTION
**Pull request by michaelneale**
There's nothing people wouldn't do for the watch.

### CODE REVIEW

**Comment by markhibberd** [comment]
Nice :

File: src/cabal [hunk 1]
```diff
@@ -118,6 +120,19 @@ run_quick () {
...
+# Load up ghcid
+#
+run_watch () {
+    initialize
+    [ $# -eq 1 ] || fail "Watch requires a single argument, specifying the entry point you want to load."
+    [ -f "$1" ] || fail "The entry point does not exist."
+    [ ! -d src ] || SRC_DIRS="-isrc"
+    [ ! -d test ] || SRC_DIRS="${SRC_DIRS:-} -itest"
+    [ ! -d dist/build/autogen ] || SRC_DIRS="${SRC_DIRS:-} -idist/build/autogen"
+    ghcid -c "ghci -package-db=$(find .cabal-sandbox -name '*-packages.conf.d') ${SRC_DIRS:-} \"$1\""
```
Comment (charleso) [inline src/cabal]: Could possibly check for `ghcid` and print out instructions to install if not found?

**Comment by charleso** [comment]
:+1:

File: src/cabal (hunk 1 above)
Comment (charleso) [inline src/cabal]: Could also make a function that generates the `ghci` command string and re-use in quick and here if you feel like it. :)

Comment (michaelneale) [inline src/cabal]: Refactoring bash is one of the best things in the whole world.

Comment (domdere) [inline src/cabal]: > Refactoring bash is one of the best things in the whole world.

yeh i love it lots

**Comment by michaelneale** [comment]
@charleso helper message look ok?

**Comment by charleso** [comment]
:guitar:

**Pull request merged**